### PR TITLE
proto: add spend to `TransactionPlannerRequest`

### PR DESCRIPTION
**Pull request by TalDerei**
## Describe your changes

previously, there wasn't an need to explicitly specify a _spend_ as a user intent until non-zero fees were introduced. In that context, this proto-related modification is necessary in order to allow a user to send an _entire_ balance for a specific asset. 

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

### CODE REVIEW

File: deployments/scripts/protobuf-codegen
Comment (cratelyn) [reply deployments/scripts/protobuf-codegen:19]: it'd be nice to pull this into a separate commit, so that this commit only does what it says on the tin: add a field to `TransactionPlannerRequest`

File: proto/penumbra/penumbra/view/v1/view.proto
Comment (cratelyn) [inline proto/penumbra/penumbra/view/v1/view.proto:759]: non-blocking nit: there's a whitespace error introduced here

File: deployments/scripts/protobuf-codegen
Comment (TalDerei) [reply deployments/scripts/protobuf-codegen:19]: > why did this change? i was a little surprised to see a change to the `protobuf-codegen` logic in this commit

since `buf dep update` was [depreciated](https://github.com/bufbuild/buf/releases)

File: proto/penumbra/penumbra/view/v1/view.proto
Comment (cratelyn) [inline proto/penumbra/penumbra/view/v1/view.proto:1]: the changes here seem good to me, though i don't have all of the context. it'd be nice if information was added to the pr description about why this change to the protocol is needed, so i can learn about that.

Comment (TalDerei) [reply proto/penumbra/penumbra/view/v1/view.proto:1]: added short description, and we discussed a bit offline

**Comment by TalDerei** [comment]
@hdevalence is this sufficient to unblock web?

**Review by erwanor — APPROVED** [review]

File: deployments/scripts/protobuf-codegen
Comment (dynst) [reply deployments/scripts/protobuf-codegen:19]: But your link says the opposite of that.

https://github.com/bufbuild/buf/releases/tag/v1.32.0

> - Move `buf mod update` to `buf dep update`. `buf mod update` is now deprecated.

Comment (TalDerei) [reply deployments/scripts/protobuf-codegen:19]: must have misread that! will fix, thanks

Comment (TalDerei) [reply deployments/scripts/protobuf-codegen:19]: > But your link says the opposite of that.
> 
> https://github.com/bufbuild/buf/releases/tag/v1.32.0
> 
> > * Move `buf mod update` to `buf dep update`. `buf mod update` is now deprecated.

https://github.com/penumbra-zone/penumbra/pull/4626

Comment (cratelyn) [reply deployments/scripts/protobuf-codegen:19]: i'm disappointed that this review feedback wasn't addressed before merging. in the future, we should resolve our peers' comments before merging code into `main`.

Comment (TalDerei) [reply deployments/scripts/protobuf-codegen:19]: > i'm disappointed that this review feedback wasn't addressed before merging. in the future, we should resolve our peers' comments before merging code into `main`.

yes, my mistake! I accidentally misread the tagged released, but I opened another PR to fix this. I will be extra cautious going forward

**Pull request merged**
